src/ui/Card.tsx

```diff
@@ -175,10 +179,22 @@ export function Card(props: CardProps): JSX.Element {
     (async (): Promise<void> => {
       const messageIds = (await fetchMessageIdsAndLabels(props.threadId))
         .messages;
+
       if (messageIds !== undefined) {
         setMessages(messageIds.map((x) => new Message(x)));
         const messageIdsToFetch = [defined(messageIds[0].id)];
         if (messageIds.length > 1) {
+          let hasMessageInInbox = false;
+          for (const messageId of messageIds) {
+            if (messageId.labelIds?.includes('INBOX')) {
+              hasMessageInInbox = true;
+              break;
+            }
+          }
+          if (!hasMessageInInbox) {
+            await modifyThread(props.threadId, [], ['INBOX']);
+            props.onCardOffScreen({removeThreadId: props.threadId});
```

Comment:
This is gross, but if this logic lives here, I think this is what we need to do?